src/components/PopupWithForm.js

```diff
@@ -7,6 +7,8 @@ export class PopupWithForm extends Popup {
         this._submitHandler = submitHandler;
         this._inputFields = 
             Array.from(this.form.querySelectorAll(".modal__input"));
+        this._submitButton = this._element.querySelector(".modal__save-button");
+        this._buttonText = this._submitButton.textContent;
     }
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`